File: VB/Models/Customer.vb
Imports DevExpress.Xpo

Public Class Customer
	Inherits XPObject

	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	Private fName As String
	Public Property Name() As String
		Get
			Return fName
		End Get
		Set(ByVal value As String)
			SetPropertyValue("Name", fName, value)
		End Set
	End Property

	<Association("Customer-Orders")>
	Public ReadOnly Property Orders() As XPCollection(Of Order)
		Get
			Return GetCollection(Of Order)("Orders")
		End Get
	End Property
End Class

File: CS/Models/Customer.cs
using DevExpress.Xpo;

public class Customer :XPObject {
    public Customer(Session session) : base(session) { }

    private string fName;
    public string Name {
        get { return fName; }
        set { SetPropertyValue("Name", ref fName, value); }
    }

    [Association("Customer-Orders")]
    public XPCollection<Order> Orders {
        get { return GetCollection<Order>("Orders"); }
    }
}